File: TerritoriesAssignment.Core/Entities/Setting/SettingValue.cs
namespace TerritoriesAssignment.Core.Entities.Setting {
	public class SettingValue : BaseObject {
		public Setting Setting { get; set; }
		public int IntegerValue { get; set; }
		public double DoubleValue { get; set; }
		public bool BooleanValue { get; set; }
	}
}

File: TerritoriesAssignment.WebApp/ClientApp/app/models/setting-value.ts
import { BaseObject } from "./base-object";
import { Guid } from "guid-typescript";
import { Setting } from "./setting";
import {SettingValueType} from "./enums/setting-value-type";

export class SettingValue extends BaseObject {
	public textValue: string;
	public doubleValue: number;
	public integerValue: number;
	public boolValue: boolean;
	public dateValue: Date;
	public setting: Setting;
	public settingValueType: SettingValueType;
	constructor(public id: Guid = null) {
		super(id);
	}
}